File: src/main/java/no/nav/tag/dittNavArbeidsgiver/clients/altinn/dto/Søknadsstatus.java
package no.nav.tag.dittNavArbeidsgiver.clients.altinn.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Søknadsstatus {
    @JsonProperty("_embedded")
    public Embedded embedded;

    public String continuationtoken;

    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Embedded {
        public List<DelegationRequest> delegationRequests;
    }
}

File: src/main/kotlin/no/nav/arbeidsgiver/min_side/clients/altinn/dto/Søknadsstatus.kt
package no.nav.arbeidsgiver.min_side.clients.altinn.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class Søknadsstatus(
    @field:JsonProperty("_embedded")
    var embedded: Embedded? = null,
    var continuationtoken: String? = null,
) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Embedded(
        var delegationRequests: List<DelegationRequest>? = null
    )
}